jbpm-bpmn2/src/test/java/org/jbpm/bpmn2/ActivityTest.java

```diff
@@ -1964,4 +1966,32 @@ public void testCallActivitySkipAbortParent() throws Exception {
         assertNotNull(log);
         assertEquals("Parent process should be completed and not aborted", ProcessInstance.STATE_COMPLETED, log.getStatus().intValue());
     }
+    
+    @Test
+    public void testBusinessRuleTaskFireLimit() throws Exception {
+        KieBase kbase = createKnowledgeBaseWithoutDumper("BPMN2-BusinessRuleTaskLoop.bpmn2",
+                "BPMN2-BusinessRuleTaskInfiniteLoop.drl");
+        ksession = createKnowledgeSession(kbase);
+        ksession.insert(new Person());
+        assertThatExceptionOfType(RuntimeException.class).isThrownBy(() -> { 
+            ksession.startProcess("BPMN2-BusinessRuleTask");
+        })
+        .withMessageContaining("Fire rule limit reached 10000");        
+    }
+    
+    @Test
+    public void testBusinessRuleTaskFireLimitAsParameter() throws Exception {
+        KieBase kbase = createKnowledgeBaseWithoutDumper("BPMN2-BusinessRuleTaskWithDataInputLoop.bpmn2",
+                "BPMN2-BusinessRuleTaskInfiniteLoop.drl");
+        ksession = createKnowledgeSession(kbase);
+        ksession.insert(new Person());
+        
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("limit", 5);
```

Comment:
You might consider using `Collections.singletonMap()` for slightly nicer code.

Reply:
matter of preference :)